src/engine/image.cpp

```diff
@@ -790,7 +790,7 @@ namespace fheroes2
         const uint8_t * gamePalette = getGamePalette();
 
         if ( flip ) {
-            const int32_t offsetInY = inY * widthIn + widthIn - 1 - inX;
+            const int32_t offsetInY = inY * widthIn + width - 1 + inX;
```

Comment:
It is better to revert this logic as these function is widely used in the code and we will need to correct many places in the code then.
Also the original logic here was made to make the same calculation for flipped and non-flipped sprites render regions.
```suggestion
            const int32_t offsetInY = inY * widthIn + widthIn - 1 - inX;
```